pkg/scalers/elasticsearch_scaler.go

```diff
@@ -0,0 +1,277 @@
+		}
+	} else {
+		meta.unsafeSsl = defaultUnsafeSsl
+	}
+
+	if val, ok := config.AuthParams["username"]; ok {
+		meta.username = val
+	} else if val, ok := config.TriggerMetadata["username"]; ok {
+		meta.username = val
+	}
+
+	if config.AuthParams["password"] != "" {
+		meta.password = config.AuthParams["password"]
+	} else if config.TriggerMetadata["passwordFromEnv"] != "" {
+		meta.password = config.ResolvedEnv[config.TriggerMetadata["passwordFromEnv"]]
+	}
+
+	index, err := GetFromAuthOrMeta(config, "index")
+	if err != nil {
+		return nil, err
+	}
+	meta.indexes = splitAndTrimBySep(index, ";")
+
+	meta.searchTemplateName, err = GetFromAuthOrMeta(config, "searchTemplateName")
+	if err != nil {
+		return nil, err
+	}
+
+	if val, ok := config.TriggerMetadata["parameters"]; ok {
+		meta.parameters = splitAndTrimBySep(val, ";")
+	}
+
+	meta.valueLocation, err = GetFromAuthOrMeta(config, "valueLocation")
+	if err != nil {
+		return nil, err
+	}
+
+	targetValue, err := GetFromAuthOrMeta(config, "targetValue")
+	if err != nil {
+		return nil, err
+	}
+	meta.targetValue, err = strconv.Atoi(targetValue)
+	if err != nil {
+		return nil, fmt.Errorf("targetValue parsing error %s", err.Error())
+	}
+
+	return &meta, nil
+}
+
+// newElasticsearchClient creates elasticsearch db connection
+func newElasticsearchClient(meta *elasticsearchMetadata) (*elasticsearch.Client, error) {
+	config := elasticsearch.Config{Addresses: meta.addresses}
+	if meta.username != "" {
+		config.Username = meta.username
+	}
+	if meta.password != "" {
+		config.Password = meta.password
+	}
+
+	if meta.unsafeSsl {
+		tr := &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		}
+		config.Transport = tr
+	}
+
+	esClient, err := elasticsearch.NewClient(config)
+	if err != nil {
+		elasticsearchLog.Error(err, fmt.Sprintf("Found error when creating client: %s", err))
+		return nil, err
+	}
+
+	_, err = esClient.Info()
+	if err != nil {
+		elasticsearchLog.Error(err, fmt.Sprintf("Found error when pinging search engine: %s", err))
+		return nil, err
+	}
+	return esClient, nil
+}
+
+func (s *elasticsearchScaler) Close(ctx context.Context) error {
+	return nil
+}
+
+// IsActive returns true if there are pending messages to be processed
+func (s *elasticsearchScaler) IsActive(ctx context.Context) (bool, error) {
+	messages, err := s.getQueryResult()
+	if err != nil {
+		elasticsearchLog.Error(err, fmt.Sprintf("Error inspecting elasticsearch: %s", err))
+		return false, err
+	}
+	return messages > 0, nil
+}
+
+// getQueryResult returns result of the scaler query
+func (s *elasticsearchScaler) getQueryResult() (int, error) {
+	// Build the request body.
+	var body bytes.Buffer
+	if err := json.NewEncoder(&body).Encode(buildQuery(s.metadata)); err != nil {
+		elasticsearchLog.Error(err, "Error encoding query: %s", err)
+	}
+
+	// Run the templated search
+	res, err := s.esClient.SearchTemplate(
+		&body,
+		s.esClient.SearchTemplate.WithIndex(s.metadata.indexes...),
```

Comment:
You should pass the context until here because you could add `.WithContext`. With this change, the request context can be managed from outside the scaler